synapse/lib/coro.py

```diff
@@ -148,6 +148,19 @@ async def ornot(func, *args, **kwargs):
         return await retn
     return retn
 
+bgtasks = set()
```

Comment:
Is this just to keep a reference ot the task so it runs?

Reply:
Yep. Otherwise it can be ripped out of the ioloop.